packages/@headlessui-react/src/components/combobox/combobox.tsx

```diff
@@ -1672,14 +1672,20 @@ function OptionsFn<TTag extends ElementType = typeof DEFAULT_OPTIONS_TAG>(
   }
 
   let [floatingRef, style] = useFloatingPanel(anchor)
+  let [localOptionsElement, setLocalOptionsElement] = useState<HTMLElement | null>(null)
```

Comment:
We should definitely add a comment about _why_ we're using this one and not the one in the parent state. Ditto for the rest of the components.